libraries/chain/controller.cpp

```diff
@@ -2825,6 +2826,34 @@ struct controller_impl {
       }
    } FC_CAPTURE_AND_RETHROW() } /// apply_block
 
+   void create_and_send_vote_msg(const block_header_state& bhs) {
+#warning use decide_vote() for strong after it is implementd by https://github.com/AntelopeIO/leap/issues/2070
+      bool strong = true;
+
+      // A vote is created and signed by each finalizer configured on the node that
+      // in active finalizer policy
+#warning possible optimization: when setting finalizer policy, update finalizer_keys_on_the_node to indicate if it is active to save going over bhs.finalizer_policy->finalizers
```

Comment:
Not sure how that would work. The active finalizer_policy will constantly be changing.

Reply:
Will remove the warning.